skiko/src/commonMain/kotlin/org/jetbrains/skia/Typeface.kt

```diff
@@ -339,11 +339,21 @@ class Typeface internal constructor(ptr: NativePointer) : RefCnt(ptr) {
      * @return  all of the family names specified by the font
      */
     val familyNames: Array<FontFamilyName>
-        get() = try {
-            Stats.onNativeCall()
-            _nGetFamilyNames(_ptr)
-        } finally {
-            reachabilityBarrier(this)
+        get() {
+            var arrayDecoder: ArrayDecoder? = null
+            return try {
+                Stats.onNativeCall()
+                arrayDecoder = ArrayDecoder(_nGetFamilyNames(_ptr), ManagedString_nGetFinalizer())
+                val size = arrayDecoder.size
+                (0 until size / 2).map { i ->
+                    val name = arrayDecoder.release(2 * i)
+                    val language = arrayDecoder.release(2 * i + 1)
+                    FontFamilyName(ManagedString(name).toString(), ManagedString(language).toString())
+                }.toTypedArray()
+            } finally {
+                arrayDecoder?.dispose()
```

Comment:
Why not use `interopScope` or create similar `arrayDecoderScope`?

Reply:
regarding the fact that we will using this in quite many places yep, make sense, pushed